ipod/utils.py

```diff
@@ -356,3 +356,165 @@ def assign_duplicate_observations(
+    orbits_initial: FittedOrbits,
+    orbits_me: FittedOrbits,
+    members_me: FittedOrbitMembers,
+) -> Tuple[MergeSummary, MERecoverySummary, list[str]]:
+    """
+    Analyze the output of the merge and extend process. This function will return
+    summaries of merged orbits, orbits that were removed, and the results of ME
+    run based on an expected set of linkages (members_expected).
+    """
+
+    # first find the merged/removed orbits
+    merge_summary, orbits_removed = identify_merged_dropped_orbits(
+        members_initial, orbits_initial, members_me, orbits_me
+    )
+
+    # now find the deltas from the expected members
+    me_results: Dict[str, List[Union[str, int, None]]] = {
+        "old_orbit_id": [],
+        "merged_orbit_id": [],
+        "expected_num_obs": [],
+        "number_observations_matched": [],
+        "total_obs_resulting_linkage": [],
+    }
+    for orb_id in members_expected.orbit_id.unique().to_pylist():
+        orbit_members = members_expected.where(
+            pc.equal(members_expected.column("orbit_id"), orb_id)
+        )
+        obs_ids = orbit_members.obs_id
+        # find obs in post-me members that match these ids
+        matching_members = members_me.apply_mask(pc.is_in(members_me.obs_id, obs_ids))
+        # Case where we only match to one orbit
+        if len(matching_members.orbit_id.unique()) == 1:
+            resulting_orbit_members_complete = members_me.where(
+                pc.equal(members_me.column("orbit_id"), matching_members.orbit_id[0])
+            )
+            me_results["old_orbit_id"].append(orb_id)
+            me_results["merged_orbit_id"].append(matching_members.orbit_id[0])
+            me_results["expected_num_obs"].append(len(orbit_members))
+            me_results["number_observations_matched"].append(len(matching_members))
+            me_results["total_obs_resulting_linkage"].append(
+                len(resulting_orbit_members_complete)
+            )
+        elif len(matching_members.orbit_id.unique()) > 1:
+            # case where an expected linkage was split between several orbits
+            for new_id in matching_members.orbit_id.unique():
+                resulting_orbit_members_complete = members_me.where(
+                    pc.equal(members_me.column("orbit_id"), new_id)
+                )
+                me_results["old_orbit_id"].append(orb_id)
+                me_results["merged_orbit_id"].append(new_id)
+                me_results["expected_num_obs"].append(len(orbit_members))
+                me_results["number_observations_matched"].append(
+                    len(
+                        matching_members.where(
+                            pc.equal(matching_members.column("orbit_id"), new_id)
+                        )
+                    )
+                )
+                me_results["total_obs_resulting_linkage"].append(
+                    len(resulting_orbit_members_complete)
+                )
```

Comment:
I think we don't need the conditional and can just use the logic where we have more than one unique orbit id. We will also need to control for when 0 orbit members match the expected orbit id.